NEW Remove aria-hidden to toggle the visibility of the Site banner

mikenuguid opened the pull request:
- hidden elements by default will be skipped by screen readers so it is not necessary to add an
`aria-hidden` attribute to show/hide active site banners. Instead we can utilise pure CSS to toggle the
display of site banners

- hide the banner using a css class instead of removing it when dismiss is clicked

- modify docs to remove the aria-hidden attribute from the template

mikenuguid committed:
NEW Remove aria-hidden to toggle the visibility of the Site banner

- hidden elements by default will be skipped by screen readers so it is not necessary to add an
aria-hidden attribute to show/hide active site banners. Instead we utilise pure CSS to toggle the
display of site banners

- hide the banner using a css class instead of removing it when dismiss is clicked

- modify docs to remove the aria-hidden attribute from the template

scott1702 commented on README.md:
```diff
@@ -43,7 +43,7 @@ The site banner can be configured in `admin/settings` now.
 In order to show the banners, you need to add them to your template:
 
 	<% loop $SiteConfig.SiteBanners %>
-        <div id="site-banner-$ID" class="site-banner site-banner-$Type" role="alert" data-id="$ID" aria-hidden="true">
```
We should also add the `site-banner--hidden` class in this example so the banners are hidden by default until the javascript runs

scott1702 reviewed (changes requested):
At the moment if we merged this in it would break backwards compatibility of the module because site banners which have previously been dismissed will be displayed by default (Because the templates in the codebases won't have the `site-banner--hidden` class on them)

One option to get around this could be have site-banners hidden by default:

```css
.site-banner {
  display: none
}

.site-banner--visible {
    display: block;
}
```

```js
    // Display the banner.
    bannersNodeList[index].classList.add('site-banner--visible');
```

Although this may also have backwards compatibility issues if consumers of the module have some css to set the site-banner to `display: flex;` for example. 🤔  Not sure how to go about getting this one through. The module hasn't been tagged before so would be dangerous to merge this into master. I'll tag the module for now and we can come back to this with a plan for how to get this change in 😅

mikenuguid commented:
Thanks for the feedback 😄 I don't mind closing this PR because based on what I observed, projects often customise the sitebanner so this can be done as a customised approach rather than a general one. But yeah, happy to go with a better plan if needed.

mikenuguid commented:
> At the moment if we merged this in it would break backwards compatibility of the module because site banners which have previously been dismissed will be displayed by default (Because the templates in the codebases won't have the `site-banner--hidden` class on them)
> 
> One option to get around this could be have site-banners hidden by default:
> 
> ```css
> .site-banner {
>   display: none
> }
> 
> .site-banner--visible {
>     display: block;
> }
> ```
> 
> ```js
>     // Display the banner.
>     bannersNodeList[index].classList.add('site-banner--visible');
> ```
> 
> Although this may also have backwards compatibility issues if consumers of the module have some css to set the site-banner to `display: flex;` for example. thinking Not sure how to go about getting this one through. The module hasn't been tagged before so would be dangerous to merge this into master. I'll tag the module for now and we can come back to this with a plan for how to get this change in sweat_smile

One idea would be to add the `.site-banner--hidden` instead of removing the aria-hidden from the css implementation. This way it's still backwards compatible but the user has the option to override the template (and script) and not use the aria-hidden attribute.

Something like:

```
.site-banner[aria-hidden='true'],
.site-banner--hidden {
    display: none;
}
```

mikenuguid committed:
NEW Update the way banners are hidden to be backward compatible

- Allow site-banners to be backward compatible by utilising both css selector and aria-hidden attribute
- Baners are now hidden (`display: none`) instead of removing them from the DOM
- Set aria-hidden="false" if banners are not hidden and aria-hidden="true" if hidden
- Update ReadME docs

satrun77 committed:
Merge branch 'master' into pulls/site-banner-accessibility
